src/robot/running/status.py

```diff
@@ -168,6 +168,7 @@ def __init__(self, parent, test, skip_on_failure=None, critical_tags=None,
         super().__init__(parent)
         self._test = test
         self._skip_on_failure_tags = skip_on_failure
+        self._skip_on_failure_tags.append('robot:skip-on-failure')
```

Comment:
`skip_on_failure` can be `None` (that's the default!) or a tuple so using `append` isn't safe. Two lines above could be replaced with this:
```python
self._skip_on_failure_tags = tuple(skip_on_failure or ()) + ('robot:skip-on-failure',)
```